mongodbatlas/data_source_mongodbatlas_projects.go

```diff
@@ -242,23 +242,28 @@ func flattenProjects(ctx context.Context, client *MongoDBClient, projects []*mat
 				diags = append(diags, diagWarning)
 			}
 
-			results[k] = map[string]interface{}{
-				"id":            project.ID,
-				"org_id":        project.OrgID,
-				"name":          project.Name,
-				"cluster_count": project.ClusterCount,
-				"created":       project.Created,
-				"teams":         flattenTeams(teams),
-				"api_keys":      flattenAPIKeys(apiKeys),
-				"is_collect_database_specifics_statistics_enabled": projectSettings.IsCollectDatabaseSpecificsStatisticsEnabled,
-				"is_data_explorer_enabled":                         projectSettings.IsDataExplorerEnabled,
-				"is_extended_storage_sizes_enabled":                projectSettings.IsExtendedStorageSizesEnabled,
-				"is_performance_advisor_enabled":                   projectSettings.IsPerformanceAdvisorEnabled,
-				"is_realtime_performance_panel_enabled":            projectSettings.IsRealtimePerformancePanelEnabled,
-				"is_schema_advisor_enabled":                        projectSettings.IsSchemaAdvisorEnabled,
-				"region_usage_restrictions":                        project.RegionUsageRestrictions,
-				"limits":                                           flattenLimits(limits),
+			resultEntry := map[string]interface{}{
+				"id":                        project.ID,
+				"org_id":                    project.OrgID,
+				"name":                      project.Name,
+				"cluster_count":             project.ClusterCount,
+				"created":                   project.Created,
+				"region_usage_restrictions": project.RegionUsageRestrictions,
+				"teams":                     flattenTeams(teams),
+				"api_keys":                  flattenAPIKeys(apiKeys),
+				"limits":                    flattenLimits(limits),
 			}
+
+			if projectSettings != nil {
+				resultEntry["is_collect_database_specifics_statistics_enabled"] = projectSettings.IsCollectDatabaseSpecificsStatisticsEnabled
```

Comment:
[q] just double-checking as I don't have full context, is it OK to not fill this data? do we need to fill empty fields by any chance?

Reply:
no, we don't have to. This is what is stored in the terraform state file which does not require to fill all the fields of the resource